pytest/tools/prober/prober_split.py

```diff
@@ -0,0 +1,156 @@
+
+    if legacy_genesis_height != split_genesis_height:
+        logger.error(
+            "The genesis height is different. legacy: {}, split {}",
+            legacy_genesis_height,
+            split_genesis_height,
+        )
+        sys.exit(1)
+
+    return legacy_genesis_height
+
+
+def check_head(legacy_url: str, split_url: str, genesis_height: int) -> int:
+    legacy_head = get_head(legacy_url)
+    split_head = get_head(split_url)
+
+    if legacy_head <= genesis_height:
+        logger.error(
+            '{} head must be higher than genesis. Got {} and {}',
+            legacy_url,
+            legacy_head,
+            genesis_height,
+        )
+        sys.exit(1)
+
+    if split_head <= genesis_height:
+        logger.error(
+            '{} head must be higher than genesis. Got {} and {}',
+            split_url,
+            split_head,
+            genesis_height,
+        )
+        sys.exit(1)
+
+    return min(legacy_head, split_head)
+
+
+def check_blocks(legacy_url: str, split_url: str, height: int):
+    logger.info(f"Checking blocks at height {height}")
+
+    legacy_block = get_block(height, legacy_url)
+    split_block = get_block(height, split_url)
+
+    if legacy_block != split_block:
+        logger.error(
+            f"Check failed, the legacy block and the split block are different",
+            f"\nlegacy block\n{pretty_print(legacy_block)}"
+            f"\nsplit block\n{pretty_print(split_block)}")
+        sys.exit(1)
+
+    return legacy_block
+
+
+def check_chunks(legacy_url: str, split_url: str, block):
+    if block is None:
+        return
+
+    logger.info(f"Checking chunks")
+
+    for chunk in block['chunks']:
+        legacy_chunk = get_chunk(chunk, legacy_url)
+        split_chunk = get_chunk(chunk, split_url)
+
+        if legacy_chunk != split_chunk:
+            logger.error(
+                f"Check failed, the legacy chunk and the split chunk are different",
+                f"\nlegacy block\n{pretty_print(legacy_chunk)}"
+                f"\nsplit block\n{pretty_print(split_chunk)}")
+            sys.exit(1)
+
+
+def main():
+    start_time = datetime.now()
+
+    parser = argparse.ArgumentParser(
+        description='Run a prober for split archival nodes')
+    parser.add_argument('--legacy-url', required=True)
+    parser.add_argument('--split-url', required=True)
+    parser.add_argument('--duration_ms', default=2000, type=int)
+    parser.add_argument('--log-level', default="INFO")
+    args = parser.parse_args()
+
+    logger.setLevel(args.log_level)
+    logger.info('Running Prober')
+
+    legacy_url = args.legacy_url
+    split_url = args.split_url
+    duration = timedelta(milliseconds=args.duration_ms)
+
+    genesis_height = check_genesis(legacy_url, split_url)
+    head = check_head(legacy_url, split_url, genesis_height)
+    logger.info(
+        f"The genesis height is {genesis_height}. The head height is {head}")
+
+    count = 0
+    none_count = 0
+    while True:
```

Comment:
Can you please comment why this loop is essential to your prober?
What if this prober does exactly one iteration?

Reply:
I wouldn't say it's essential but it seemed a waste to check only one block in one execution of the prober. I'm thinking about having it run every 60s and setting the duration to 50s so that it's actually doing something most of the time. I will add a comment.